tofu/geom/_core_optics.py

```diff
@@ -1938,14 +2067,33 @@ def plot_focal_error_summed(
                 print(msg, end=end, flush=True)
 
                 # Get det
-                det = self.get_detector_approx(
-                    ddist=ddist[ii],
-                    di=di[jj],
-                    lamb=lamb,
-                    bragg=bragg,
-                    use_non_parallelism=use_non_parallelism,
-                    tangent_to_rowland=tangent_to_rowland,
-                )
+                if not tangent_to_rowland:
+                    det = self.get_detector_approx(
+                        ddist=ddist[ii],
+                        di=di[jj],
+                        dj=dj0,
+                        dtheta=dtheta0,
+                        dpsi=dpsi0,
+                        tilt=tilt0,
+                        lamb=lamb,
+                        bragg=bragg,
+                        use_non_parallelism=use_non_parallelism,
+                        tangent_to_rowland=False,
+                    )
+                else:
+                    det = self.get_detector_approx(
+                        ddist=ddist[ii],
+                        di=di[jj],
+                        dj=dj0,
+                        # need to re-adapt these rotations to tangent case
+                        dtheta=dtheta0,
+                        dpsi=dpsi0-angle_nout,
+                        tilt=tilt0,
+                        lamb=lamb,
+                        bragg=bragg,
+                        use_non_parallelism=use_non_parallelism,
+                        tangent_to_rowland=True,
+                    )
```

Comment:
Shorter formulation:

                psi0bis = float(psi0)
                if tangent_to_rowland:
                    psi0bis = psi0 - angle_nout
                det = self.get_detector_approx(
                    ddist=ddist[ii],
                    di=di[jj],
                    dj=dj0,
                    dtheta=dtheta0,
                    dpsi=dpsi0bis,
                    tilt=tilt0,
                    lamb=lamb,
                    bragg=bragg,
                    use_non_parallelism=use_non_parallelism,
                    tangent_to_rowland=tangent_to_rowland,
                )

Reply:
Thanks for that too.